### ISSUE TITLE
Cannot ❌ use emoji as migration name 😨

### ISSUE THREAD

**Issue by vegerot** [typeorm/typeorm#5581]
**Issue type:**

[ ] question
[ x] 🐛 report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x ] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ x] `latest`

**Steps to reproduce or a small repository showing the problem:**
Trying to make a migration with an emoji in the name succeeds.  However, running the migrations comes out with tons of errors (about 1100 lines depending on the size of the migration).  Looking into the generated migration file more it appears that the generated class is invalid.  TypeORM should either add support for emoji (which seems [unlikely](https://mathiasbynens.be/notes/javascript-identifiers-es6)) or throw an error while generating, to not make it appear to have generated successfully.

Example:
```
> ts-node -r tsconfig-paths/register ./node_modules/typeorm/cli.js "migration:generate" "-n" "😀"

Migration /Users/maxcoplan/api/src/migration/1582850804068-😀.ts has been generated successfully.
```

plz fix

P.S. 🤡

**Comment by artoodeeto** [comment]
do you have any use case for this? why would you name ur migration as emoji? just wondering. :thinking:

**Comment by vegerot** [comment]
@artoodeeto no use case, just noticed the bug when I was bored.  I'm not saying we should add support for emoji, I'm saying an error should be thrown when generating the migration file with unsupported unicode symbols because TypeScript can't run it.

**Comment by artoodeeto** [comment]
@vegerot oh yea they should throw an error on this but I dont think that this is important. hehe. but niche catch though. :smile:

**Comment by vegerot** [comment]
@artoodeeto agreed, I don't think it is important either.  But it is a low-priority bug that SHOULD be fixed (very very low priority)

**Comment by vegerot** [comment]
I still think this is worth fixing @imnotjames
